odp_sdk/tabular_storage_client.py

```diff
@@ -318,28 +330,7 @@ def select_as_list(
             OdpResourceNotFoundError: If the schema cannot be found
         """
 
-        row_iterator = self._select_stream(resource_dto, filter_query, limit, cursor)
-
-        if limit:
-            return list(row_iterator)
-        else:
-            return list(row_iterator)[:-1]
-
-    def _select_stream(
-        self,
-        resource_dto: ResourceDto,
-        filter_query: Optional[dict] = None,
-        limit: Optional[int] = None,
-        cursor: Optional[str] = None,
-    ) -> Iterator[dict]:
-        """
-        Helper method to get data in chunks and to compile them
-        """
-        # Make sure the page size is a multiple of 1000 per API specs
```

Comment:
As we remove this limit recalculation, it will not be needed to reintroduce as per tabular API specs?

Reply:
Good point. Introduced an alignment factor:

```python
if use_limit % self.page_size_align_factor:
    # Round up to the nearest multiple of the alignment factor
    use_limit = (use_limit // self.page_size_align_factor + 1) * self.page_size_align_factor
```